.github/PULL_REQUEST_TEMPLATE/pull-request-template.md

```diff
@@ -0,0 +1,25 @@
+# Change Summary
+
+## Overview
+Add a list or paragraph giving an overview of your changes
+
+## New Dependencies
+List any new dependencies introduced by these changes
+
+## New Files
+- new file 1
+   - description of new file 1's purpose
+
+## Deleted Files
+- deleted file 1
+   - explanation for why file was deleted
+
+## Updated Files
+- updated file 1
+   - description of change 1 in file 1
+   - description of change 2 in file 2
+- updated file 2
+   - descipriton of change 1 in file 2
```

Comment:
```suggestion
```
I am not sure this much detail is needed unless you want to draw a reviewer's attention to something specific because these are all shown in the Files section of the PR, so it seems like more work for the author to list specific updates like this.

Sometimes if I want to draw attention to something specific I'll review my own PR and leave a comment at that line for the specific callout too.

Reply:
actually, reading through your review that uses this format, maybe it is useful to have the more detailed template, which then can be slimmed down as needed. I guess if you find yourself writing a really long description, that might be a sign more than one PR is needed.

Reply:
I think the benefit to the file detail is it saves the reviewer time and better prepares them when they look at the file, since they will already know the purpose of those changes to that specific file instead of having to interpret it by reading the code. Plus the description provides an extra check to confirm that what the author thinks the changes are doing line up with what the reviewer sees the code doing. I can add a note that those bullets shouldn't be describing every line change, just the larger concepts behind each change. It is more work for the author, but it shouldn't take more than a few minutes unless your PR is way too large. Does anyone else have an opinion on this?